packages/java/engine-core/src/main/java/com/vaadin/hilla/engine/EngineConfiguration.java

```diff
@@ -21,319 +22,412 @@
+    }
 
-        public Builder generator(GeneratorConfiguration value) {
-            configuration.generator = value;
-            return this;
-        }
+    default EngineConfiguration setOutputDir(String outputDir) {
+        setOutputDir(getBaseDir().resolve(outputDir));
+        return this;
+    }
 
-        public Builder outputDir(String value) {
-            return outputDir(Path.of(value));
-        }
+    default EngineConfiguration setOutputDir(Path outputDir) {
+        State.INSTANCE.outputDir = outputDir;
+        return this;
+    }
 
-        public Builder outputDir(Path value) {
-            configuration.outputDir = resolve(value);
-            return this;
-        }
+    default EngineConfiguration setParser(ParserConfiguration parser) {
+        State.INSTANCE.parser = parser;
+        return this;
+    }
 
-        public Builder parser(ParserConfiguration value) {
-            configuration.parser = value;
-            return this;
-        }
+    default EngineConfiguration setGroupId(String groupId) {
+        State.INSTANCE.groupId = groupId;
+        return this;
+    }
 
-        public Builder groupId(String value) {
-            configuration.groupId = value;
-            return this;
-        }
+    default EngineConfiguration setArtifactId(String artifactId) {
+        State.INSTANCE.artifactId = artifactId;
+        return this;
+    }
 
-        public Builder artifactId(String value) {
-            configuration.artifactId = value;
-            return this;
-        }
+    default EngineConfiguration setMainClass(String mainClass) {
+        State.INSTANCE.mainClass = mainClass;
+        return this;
+    }
 
-        public Builder mainClass(String value) {
-            configuration.mainClass = value;
-            return this;
-        }
+    default EngineConfiguration setBrowserCallableFinders(
+            BrowserCallableFinder... browserCallableFinders) {
+        State.INSTANCE.browserCallableFinders = Arrays
+                .asList(browserCallableFinders);
+        return this;
+    }
 
-        public Builder browserCallableFinder(BrowserCallableFinder finder) {
-            configuration.browserCallableFinder = finder;
-            return this;
-        }
+    default List<Class<?>> findBrowserCallables()
+            throws ExecutionFailedException {
+        var iterator = getBrowserCallableFinders().iterator();
 
-        public Builder productionMode(boolean value) {
-            configuration.productionMode = value;
-            return this;
-        }
+        while (iterator.hasNext()) {
+            var finder = iterator.next();
 
-        public Builder nodeCommand(String value) {
-            configuration.nodeCommand = value;
-            return this;
+            try {
+                return finder.find(this);
+            } catch (ExecutionFailedException e) {
+                if (iterator.hasNext()) {
+                    LOGGER.debug("Failed to find browser-callables", e);
```

Comment:
I'd suggest to enhance this log message, since it might be confusing to see a "Failed to find browser-callables" message and a stacktrace, while the other finder succeeds and the app is working.